default-fields/0000-default-fields.md

```diff
@@ -6,19 +6,224 @@
 # Summary
 [summary]: #summary
 
-One para explanation of the feature.
+Allow struct definitions to supply default values for individual fields,
+and then allow those fields to be omitted from construction:
+
+```rust
+struct Foo {
+    a: &'static str,
+    b: bool = true,
+    c: i32
+}
+
+let foo = Foo {
+    a: "Hello",
+    c: 42
+}
+```
 
 # Motivation
 [motivation]: #motivation
 
-Why are we doing this? What use cases does it support? What is the expected outcome?
+Today, Rust allows you to create an instance of a struct using a literal syntax:
+
+```rust
+struct Foo {
+    a: &'static str,
+    b: bool,
+    c: i32
+}
+
+let foo = Foo {
+    a: "Hello",
+    b: true,
+    c: 42
+};
+```
+
+This requires all fields in the struct be assigned a value, unless the struct implements
+the Default trait.
+In this case, values can be optionally provided, and rest assigned their default value:
+
+```rust
+#[derive(Default)]
+struct Foo {
+    a: &'static str,
+    b: bool,
+    c: i32
+}
+
+let foo = Foo {
+    a: "Hello",
+    c: 42,
+    ..Default::default()
+}
+```
+
+The problem with this is that it requires Foo to derive Default, which may not always
+be desirable.
+
+The existing struct construction also can't be used when a struct has private fields:
+
+```rust
+mod data {
+    pub struct Foo {
+        pub a: &'static str,
+        b: bool,
+        pub c: i32
+    }
+}
+
+// Error: field `b` is private
+let foo = data::Foo {
+    a: "Hello",
+    b: true,
+    c: 42
+}
+```
+
+To work around these shortcomings, users can create constructor functions or more elaborate builders.
+The problem with a constructor is that it needs to be kept in sync with the fields of the struct, or
+require the user reassign values they want to change.
```

Comment:
I'd say a bigger problem with ctors is you need one for each combination of fields